services/matrix.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+var v = require('./service-helpers/validator');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+var pick = require('./service-helpers/pick');
+
+/**
+ * Map Matching API service.
+ */
+var Matrix = {};
+
+/**
+ * Returns a duration and/or distance matrix showing travel times and distances between coordinates.
+ *
+ * See the [Mapbox Direction Matrix API](https://www.mapbox.com/api-documentation/#matrix).
+ *
+ * @param {Object} config
+ * @param {Array<MatrixPath>} config.matrixPath - An ordered array of object containing coordinates and related properties. The size of this array must be between 2 & 100 (inclusive).
+ * @param {'driving-traffic'|'driving'|'walking'|'cycling'} [config.profile=driving]
+ * @param {Array<'all'|number>} [config.sources] - Use coordinates with given index as sources.
+ * @param {Array<'all'|number>} [config.destinations] - Use coordinates with given index as destinations.
+ * @param {Array<'distance'|'duration'>} [config.annotations] - Whether or not to return additional metadata along the route.
+ * @return {MapiRequest}
+ */
+Matrix.getMatrix = function(config) {
+  v.assertShape({
+    matrixPath: v.required(
+      v.arrayOf(
+        v.shape({
+          coordinates: v.required(v.coordinates),
+          approach: v.oneOf('unrestricted', 'curb')
+        })
+      )
+    ),
+    profile: v.oneOf('driving-traffic', 'driving', 'walking', 'cycling'),
+    annotations: v.arrayOf(v.oneOf('duration', 'distance')),
+    sources: v.oneOfType(v.equal('all'), v.arrayOf(v.number)),
+    destinations: v.oneOfType(v.equal('all'), v.arrayOf(v.number))
+  })(config);
+
+  config.profile = config.profile || 'driving';
```

Comment:
The API requires this value instead of providing a default. Do you think we should provide *additional* defaults like this in the SDK, or require all the input that the HTTP API requires?

I ask because I was thinking about this briefly for Geocoding: it would make sense to default the `mode` to `mapbox.places` (instead of `mapbox.places-permanent`), but the API didn't do that so I was hesitant.

What do you think?

Reply:
I feel the directions profile is special option, since it goes in the url. I can make it required 🤷🏽‍♀️

Reply:
Whatever you want to go with is ok with me. Let's at some point make sure that Geocoding matches the pattern.

Reply:
Went ahead and [updated](https://github.com/mapbox/mapbox-sdk-js/pull/234/commits/82f4cff2440e7c4bdf7c18535b761ed1e23f6fb7) geocoding service to use a default mode.